sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/events.scala

```diff
@@ -110,7 +120,27 @@ case class RenameTableEvent(
   extends TableEvent
 
 /**
- * Event fired when a function is created, dropped or renamed.
+ * Event fired before a table is altered.
+ */
+case class AlterTablePreEvent(database: String, name: String) extends TableEvent
+
+/**
+ * Event fired after a table is altered.
+ */
+case class AlterTableEvent(database: String, name: String) extends TableEvent
```

Comment:
Maybe we can have one event for all `alterTableXXX`, and add one more parameter to indicate which part of the table is altered.